packages/flame/lib/src/layout/align_component.dart

```diff
@@ -48,41 +48,61 @@ import 'package:vector_math/vector_math_64.dart';
 /// );
 /// ```
 class AlignComponent extends PositionComponent {
-  /// Creates a component that keeps its [child] positioned according to the
+  /// Creates a component that keeps its [_child] positioned according to the
   /// [alignment] within this component's bounding box.
   ///
   /// More precisely, the child will be placed at [alignment] relative position
   /// within the current component's bounding box. The child's anchor will also
   /// be set to the [alignment], unless [keepChildAnchor] parameter is true.
   AlignComponent({
-    required this.child,
-    required Anchor alignment,
+    PositionComponent? child,
+    Anchor alignment = Anchor.center,
     this.widthFactor,
     this.heightFactor,
     this.keepChildAnchor = false,
-  }) {
+  }) : _child = child {
     this.alignment = alignment;
-    add(child);
+    this.child = child;
   }
 
-  late Anchor _alignment;
-
-  /// The component that will be positioned by this component. The [child] will
+  /// The component that will be positioned by this component. The [_child] will
   /// be automatically mounted to the current component.
-  final PositionComponent child;
+  PositionComponent? _child;
+
+  set child(PositionComponent? value) {
+    if (_child == value) {
+      return;
+    }
+    final previousChild = _child;
+    if (previousChild?.parent == this) {
```

Comment:
I think we can connect a listener to when the child gets its parent updated if I remember correctly and then we can unassign `_child` if it is added to another parent.

Reply:
Yes, I understand the problem. Perhaps the child will change parents outside of our class?

Reply:
How was this problem solved before?

Reply:
<img width="1116" alt="Снимок экрана 2023-09-24 в 13 27 27" src="https://github.com/flame-engine/flame/assets/18004353/4620135b-2b9e-45b0-b6e2-dfb3a7182297">
I failed the golden test by 0.18%
How is that?)))

Reply:
> I failed the golden test by 0.18% How is that?))) 
It's because you're generating the goldens on an OS that is not Linux, I can generate them for you :)

Reply:
Great! What else needs to be changed in the code?

Reply:
You can override this class and set the child to null when it has been assigned to another parent:
```dart
  /// This method will be invoked from lifecycle if [child] has been added
  /// to or removed from its parent children list.
  void onChildrenChanged(Component child, ChildrenChangeType type) {}
  ```

Reply:
I understand correctly?
``` 
@override
  void onChildrenChanged(Component child, ChildrenChangeType type) {
    if (_child?.parent != this) {
      this.child = null;
    }
  }
```

Reply:
Exactly!

Reply:
I added everything to the commit
I want to use this component for a button and make a system of labels on the button depending on the states.